File: TwoSum.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;


public class TwoSum {
	public static int[] twoSum(int[] nums, int target) {
		Map<Integer, Integer> map = new HashMap<Integer, Integer>();
		int[] defaultResult = {0, 0};
		for (int i = 0; i < nums.length; i++) {
			int temp = target - nums[i];
			if (map.get(temp) != null) {
				int[] result = {map.get(temp) + 1, i + 1};
				return result;
			}
			map.put(nums[i], i);
		}
		return defaultResult;
	}
	public static void main(String[] args) {
		int[] nums = {2, 2, 7, 11, 15};
		int target = 9;
		int[] result = twoSum(nums, target);
		System.out.println(Arrays.toString(result));
	}
}

File: py_solution/two_sum.py


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        default_result = [0, 0]
        for i in range(len(nums)):
            temp = target - nums[i]
            if temp in map:
                result = [map[temp], i]
                return result
            map[nums[i]] = i
        return default_result


if __name__ == "__main__":
    solution = Solution()
    print(solution.twoSum([2, 7, 11, 15], 9))

